permission_handler_platform_interface/lib/src/method_channel/method_channel_permission_handler.dart

```diff
@@ -90,7 +90,10 @@ class MethodChannelPermissionHandler extends PermissionHandlerPlatform {
       Permission permission) async {
     final shouldShowRationale = await _methodChannel.invokeMethod(
         'shouldShowRequestPermissionRationale', permission.value);
-
+    // calling this on ios will return an int value 0, this will cause type cast error
+    if (shouldShowRationale is int){
+      return shouldShowRationale > 0;
+    }
```

Comment:
Missed this completely, however since this is very Android specific implementation it would be better to add the following at the top of the method:

```dart
    if (defaultTargetPlatform != TargetPlaform.android) {
        return false;
    }
```
This would also solve the issue and on top prevent method calls to other platforms that might not be implemented and throw an unimplemented error.